design/accepted/gc_plus/gc-plus-milestone2-execution-plan.md

```diff
@@ -0,0 +1,50 @@
+# Uncommitted Garbage Collection - Milestone 2 Execution Plan
+
+Uncommitted Garbage Collection [Proposal](https://github.com/treeverse/lakeFS/blob/master/design/accepted/gc_plus/uncommitted-gc.md)
+
+## Milestone 1
+The first beta version that was released included:
+1. Implementation of the clean run flow for old and new repository structures (without optimizations)
+2. Mark & Sweep
+3. Integration tests
+4. Backup & Restore
+
+## Milestone 2
+
+### Goals
+1. Removing the limitation of a read-only lakeFS during the job run
+2. Performance improvements - better parallelization of the storage namespace traversal
+3. Implementing Optimized Run
+
+### Non-Goals
+1. Support for non-S3 repositories
```

Comment:
This is a big bullet - I think it should be itemized by storage providers. Also I really think we should do it incremental.
For example - for this milestone, starting with Azure - and afterwards working on GCS